src/napari_imagej/java.py

```diff
@@ -118,9 +126,33 @@ def initialize(self):
         log_debug("Completed JVM Configuration")
 
         # Launch PyImageJ
-        self.ij = imagej.init(**ij_settings)
+        self.ij = (
+            imagej.gateway
+            if hasattr(imagej, "gateway") and imagej.gateway
+            else imagej.init(**ij_settings)
+        )
         # Validate PyImageJ
         self._validate_imagej()
+
+        # HACK: Avoid FlatLaf with ImageJ2 Swing UI; it doesn't work for reasons unknown.
+        try:
+            ui = self.ij.ui().getDefaultUI().getInfo().getName()
+            log_debug(f"Default SciJava UI is {ui}.")
+            if ui == "swing":
+                SwingLookAndFeelService = jimport("org.scijava.ui.swing.laf.SwingLookAndFeelService")
+                laf = self.ij.prefs().get(SwingLookAndFeelService, "lookAndFeel")
+                log_debug(f"Preferred Look+Feel is {laf}.")
+                if laf is None or laf.startsWith("FlatLaf"):
+                    UIManager = jimport("javax.swing.UIManager")
+                    fallback_laf = UIManager.getSystemLookAndFeelClassName()
+                    log_debug(f"Detected FlatLaf. Falling back to {fallback_laf} instead to avoid problems.")
+                    self.ij.prefs().put(SwingLookAndFeelService, "lookAndFeel", fallback_laf)
+        except Exception as exc:
+            from scyjava import jstacktrace
+            print(jstacktrace(exc))
+            # NB: The hack failed, but no worries, just try to keep going.
+            pass
```

Comment:
Could this be moved to the menu controls? It will only be needed if we press the button...

Reply:
That makes sense, I hadn't thought about the other ways that you could make a Swing component. Since you're the expert on this, can you just describe why it's here, then, in the `HACK` comment, so I don't try to move it later?

Reply:
> (For this same reason, I don't like how the left and right arrow buttons are grayed out until the "show GUI" button is pushed.)

Sure, I just don't have the GUI chops/experience to devise a better way. I also am not a huge fan of the icons; I'm sure there's an icon that could encapsulate the button's functionality better. If you have an idea, please file an issue!

Reply:
Comment added in 773fb2e